LinkedList/LinkedList/DoublyList.h

```diff
@@ -0,0 +1,429 @@
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+	
+
+	//---------------------------------Access
+	Type& operator[](const int Index)
+	{
+		FNode* Current;
+		if (Index < Size / 2)
+		{
+			Current = Head;
+			for (int i = 0; i < Index / 2; i++)
+			{
+				Current = Current->Next;
+			}
+			return Current->Data;
+		}
+		else
+		{
+			Current = Tail;
+			for (int i = Size - 1; i < Index / 2; i--)
+			{
+				Current = Current->Previuos;
+			}
+			return Current->Data;
+		}
+	}
+
+	const Type& operator[](const int Index) const
+	{
+		FNode* Current;
+		if (Index < Size / 2)
+		{
+			Current = Head;
+			for (int i = 0; i < Index / 2; i++)
+			{
+				Current = Current->Next;
+			}
+			return Current->Data;
+		}
+		else
+		{
+			Current = Tail;
+			for (int i = Size - 1; i < Index / 2; i--)
+			{
+				Current = Current->Previuos;
+			}
+			return Current->Data;
+		}
+	}
+
+	//-------------------------------------------------------
+	TDLList& operator=(const TDLList<Type>& OtherList)
+	{
+		Clear();
+		FNode* CurrentNode = OtherList.Head;
+
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(CurrentNode->Data);
+			CurrentNode = CurrentNode->Next;
+		}
+		return *this;
+	}
+
+	//--------------------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Data;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Data;
+	}
+
+	//---------------------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Data;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Data;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Data = InHead;
+
+		if (Head == nullptr)
+		{
+			Tail = NewHead;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head->Previuos = NewHead;
+			NewHead->Previuos = nullptr;
+		}
+		Head = NewHead;
+		Size++;
+	}
+
+	//--------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Data = InTail;
+			Tail->Next->Previuos = Tail;
+			Tail = Tail->Next;
+			
+			Size++;
+		}
+	}
+
+	//--------------------------------------------------
+	void Insert(const Type& NewData, const int Index)
+	{
+		if (Index < 0 || Index > Size)
+		{
+			return;
+		}
+		else if (Index == 0)
+		{
+			AddHead(NewData);
+		}
+		else
+		{
+			FNode* Current = GetNode(Index);
+			if (Current == Tail)
+			{
+				AddTail(NewData);
+			}
```

Comment:
This could've been placed on another if statement after the Index one, maybe one being `Index == Size`, this might help improve readability and avoid the conditional here.